atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -130,9 +130,27 @@ public NettyMessagingService(
     this.advertisedAddress = advertisedAddress;
     this.protocolVersion = protocolVersion;
     this.config = config;
+    this.channelPool = new ChannelPool(this::openChannel, config.getConnectionPoolSize());
+
+    openFutures = new CopyOnWriteArrayList<>();
+    initAddresses(config);
+  }
+
+  // duplicated for tests - to inject channel pool
```

Comment:
```suggestion
  @VisibleForTesting
```

We have this annotation as part of zeebe-util. We could extend it to add a reason/justification as well :+1: